### PR TITLE
Add desktop only redux slices

### PR DESCRIPTION
**Pull request by Caleb-T-Owens**
---
This is **part 2 of 2 in a stack** made with GitButler:
- <kbd>&nbsp;2&nbsp;</kbd> #5739 👈 
- <kbd>&nbsp;1&nbsp;</kbd> #5669

### CODE REVIEW

File: apps/desktop/src/routes/+layout.svelte [hunk 1]
```diff
@@ -64,14 +65,16 @@
...
 
-	const appState = new AppState();
+	const appState = new DesktopState();
 	const feedService = new FeedService(data.cloud, appState.appDispatch);
 	const organizationService = new OrganizationService(data.cloud, appState.appDispatch);
 	const cloudUserService = new CloudUserService(data.cloud, appState.appDispatch);
 	const cloudProjectService = new CloudProjectService(data.cloud, appState.appDispatch);
 
 	setContext(AppState, appState);
 	setContext(AppDispatch, appState.appDispatch);
```
Comment (Caleb-T-Owens) [inline apps/desktop/src/routes/+layout.svelte:75]: We still want to provide the less specific contexts so components defined in `@gitbutler/shared` can provide access the shared store

File: apps/desktop/src/lib/redux/store.svelte.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+import { usersReducer } from '@gitbutler/shared/users/usersSlice';
+import { configureStore, createSelector, createSlice } from '@reduxjs/toolkit';
+
+type DesktopOnly = {
+	value: number;
+};
+
+const desktopOnly = createSlice({
+	name: 'desktopOnly',
+	initialState: { value: 69 } as DesktopOnly,
```
Comment (mtsgrd) [inline apps/desktop/src/lib/redux/store.svelte.ts:16]: nice

File: apps/desktop/src/routes/reduxExample/+page.svelte [hunk 3]
```diff
@@ -15,15 +20,17 @@
 	 * `location = '/reduxExample'` in the console.
 	 */
 
-	const appState = getContext(AppState);
-	const appDispatch = getContext(AppDispatch);
+	const appState = getContext(DesktopState);
+	const appDispatch = getContext(DesktopDispatch);
```
Comment (Caleb-T-Owens) [inline apps/desktop/src/routes/reduxExample/+page.svelte:24]: We ought to only get the DesktopState when we actually need the desktop specific slices, so I've only updated this one component.

File: apps/desktop/tsconfig.json [hunk 4]
```diff
@@ -1,7 +1,7 @@
 {
 	"extends": "./.svelte-kit/tsconfig.json",
 	"compilerOptions": {
-		"target": "es6",
+		"target": "ES2022",
```
Comment (mtsgrd) [inline apps/desktop/tsconfig.json:4]: 👍

Comment (Caleb-T-Owens) [inline apps/desktop/tsconfig.json:4]: When targeting es6, the `$state()` and `$derived()` statements weren't allowed in classes due to how ts transformed them into declarations inside the constructor.

**Review by mtsgrd — APPROVED** [review]
I like it!

**Pull request merged**